### PR TITLE
stream: Add sort and limit params for stream

### PR DESCRIPTION
**Pull request by dsabati**
Add possibility to set a custom sort and a limit when using streams.

### CODE REVIEW

File: test/mongo-read-stream.test.ts [hunk 1]
```diff
@@ -120,6 +120,73 @@ ava('[MONGO-READ-STREAM] Create stream with hint OK', async (t: Assertions) => {
...
+		if (!item) t.fail('missing item');
+		if (lastItem && lastItem.i > item.i) t.fail('bad sorting');
+
+		lastItem = item;
+		await waitFor(2); // add some time to simulate long process
+	});
+	t.falsy(stream.query.$and);
+
+	lastItem = null;
+	stream = mongoClient.streamWithType({}, TestItem, 5, undefined, undefined, { i: -1 });
```
Comment (BenjD90) [inline test/mongo-read-stream.test.ts:145]: if you keep the sort, add a test with composed sorts

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/mongo-read-stream.ts [hunk 2]
```diff
@@ -79,13 +80,17 @@ export class MongoReadStream<
 		private readonly projection: object = {},
 		private readonly customType?: ClassType<Partial<U | L>>,
 		private readonly hint?: string | object,
+		private readonly sort: object = { _id: 1 },
```
[the remark this answers is not in the document]
Comment (dsabati) [reply src/mongo-read-stream.ts:83]: After trying a few things and giving it thoughts, is would not be possible to easily adapt the query with the sort param:

Imagine we have the following objects in our collection:
```
{ key: '1', value: '1' }
{ key: '1', value: '2' }
{ key: '2': value: '1' }
{ key: '2': value: '2' }
{ key: '3': value: '1' }
{ key: '3': value: '2' }
```

and the following sort: { key: 1, value: 1 }
and the following pagination size: 3

The first iteration would get the first 3 items without issues.
However, in the next iteration we would have the following query:
```
{
   $and: [
      { key: { $gt: '2' } },
      { value: { $gt: '1' } },
   ]
}
```
 
Which will end up only fetching the last object { key: '3': value: '2' } because it's the only meeting the query conditions.

Since I don't want to spend too much time on this, I suggest we remove the sort param and only sort by _id as it is not needed for now, what do you think ?

File: test/mongo-read-stream.test.ts (hunk 1 above)
Comment (dsabati) [reply test/mongo-read-stream.test.ts]: done!

File: src/mongo-read-stream.ts [hunk 3]
```diff
@@ -173,11 +167,54 @@ export class MongoReadStream<
...
 			}
 			this.push(item);
 		} catch (e) {
 			this.emit('error', e);
 		}
 	}
+
+	private updateQueryWithSortParams(): void {
+		this._query.$and = this._query.$and ?? [];
+		if (LodashReplacerUtils.IS_OBJECT_EMPTY(this.sort)) return;
```
Comment (BenjD90) [inline src/mongo-read-stream.ts:181]: Could you add a test with empty sort ? I think we should always sort at least by _id instead of nothing, no ?

**Review by BenjD90 — CHANGES_REQUESTED** [review]
